contracts/Shielder.sol

```diff
@@ -182,6 +189,12 @@ contract Shielder is
     {
         uint256 amount = msg.value;
         if (amount == 0) revert ZeroAmount();
+        if (
+            oldNullifierHash >= FIELD_MODULUS ||
+            newNote >= FIELD_MODULUS ||
+            idHiding >= FIELD_MODULUS
+        ) revert NotAFieldElement();
+
```

Comment:
Would try to wrap the check into a modifier/function

Reply:
I used modifier, thanks for the idea!